### ISSUE TITLE
DCAE 1.5 style latent masking

### ISSUE THREAD

**Issue by shahbuland** [Wayfarer-Labs/owl-wms#65]
DCAE 1.5 proposes several changes to standard diffusion training setup both in auto encoder and also in the diffusion model that gives a sizeable boost to performance and training speed. We are now using this modification in our autoencoder, so it seems relevant to also use the recommended modification to diffusion training. As far as the implementation goes, it seems you pick a random c' value, ala

```
@torch.no_grad()
def get_ch_prime(total_ch, min_ch=16, step=4):
    """
    Returns an integer sampled from the sequence [min_ch, min_ch+step, ..., total_ch]
    (inclusive of total_ch if it lands on a step), and ensures the value is the same across all devices.
    """
    # Build the valid channel choices
    ch_choices = list(range(min_ch, total_ch + 1, step))
    ch_choices_tensor = torch.tensor(ch_choices, dtype=torch.long, device='cuda' if torch.cuda.is_available() else 'cpu')
    num_choices = len(ch_choices)

    if torch.distributed.is_initialized():
        rank = torch.distributed.get_rank()
        if rank == 0:
            idx = torch.randint(0, num_choices, (1,), device=ch_choices_tensor.device)
        else:
            idx = torch.empty(1, dtype=torch.long, device=ch_choices_tensor.device)
        torch.distributed.broadcast(idx, src=0)
        ch_prime = ch_choices_tensor[idx.item()]
        return int(ch_prime.item())
    else:
        idx = torch.randint(0, num_choices, (1,), device=ch_choices_tensor.device)
        ch_prime = ch_choices_tensor[idx.item()]
        return int(ch_prime.item())
```

The idea is that it's sampled from [16,20,24,...,latent_channels]. The value should be the same across all devices so that for this entire batch we focus on some subset of the channels. Using this, you construct a mask that is applied during diffusion loss computation, so that the loss is only computed on the first c' channels.

Implementation should be configurable via config in the sense that it can be enabled/disabled. The GameRFT training wrapper can just use this when computing loss. Only relevant during pretraining.
